elements/map/src/generate.ts

```diff
@@ -11,23 +11,55 @@ const availableLayers = {
   STAC,
 };
 
+export type layerType =
+  | "Group"
+  | "Heatmap"
+  | "Image"
+  | "Layer"
+  | "Tile"
+  | "Vector"
+  | "VectorImage"
+  | "VectorTile";
+export type sourceType =
+  | "BingMaps"
+  | "Cluster"
+  | "GeoTIFF"
+  | "IIIF"
+  | "Image"
+  | "ImageCanvas"
+  | "ImageStatic"
+  | "ImageWMS"
+  | "OSM"
+  | "Raster"
+  | "StadiaMaps"
+  | "Tile"
+  | "TileArcGISRest"
+  | "TileDebug"
+  | "TileImage"
+  | "TileJSON"
+  | "TileWMS"
+  | "UrlTile"
+  | "Vector"
+  | "VectorTile"
+  | "WMTS"
+  | "XYZ";
+
 const availableSources = {
   ...olSources,
 };
 
 export type EoxLayer = {
-  type: olLayers.Layer;
-  id: string;
-  properties?: Object;
-  source?: { type: olSources.Source };
+  type: layerType;
+  id?: string;
```

Comment:
I think we can agree to always put the `id` into the `properties` object, to be OL API docs conformant. What do you say?

Reply:
true, i updated the EoxLayer type

Reply:
While we're at it, the `group` id should also move into the `properties` to be OL conform, currently the `group`is added at root level ike the `id` @RobertOrthofer

Reply:
[60b65c4](https://github.com/EOX-A/EOxElements/pull/220/commits/60b65c4a1172a48daf6720f0e94a368b7fa69380) now sets a reference to the parent group layer (not just the id), that can be retrieved via `.get(_group)`. Is this ok?